opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -1886,6 +1886,33 @@ namespace {
     }
 
     template<class T>
+    double
+    FullyImplicitBlackoilSolver<T>::convergenceReduction(const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& B,
+                                                         const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& tempV,
+                                                         const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& R,
+                                                         std::array<double,MaxNumPhases>& R_sum,
+                                                         std::array<double,MaxNumPhases>& maxCoeff,
+                                                         std::array<double,MaxNumPhases>& B_avg,
+                                                         int nc) const
+    {
+        const Opm::PhaseUsage& pu = fluid_.phaseUsage();
+        // Do the global reductions
+        for(int idx=0; idx<MaxNumPhases; ++idx)
```

Comment:
I think this entire loop has a problem: do you intend to fill R_sum for all possible faces (also inactive)? In that case you should not read from R.col(pu.phase_pos[idx]) and write to R_sum[pu.phase_pos[idx]] but read from R.col(idx) and write to R_sum[idx]. However, judging by the code at the call site, I think you intend to read/write only the columns corresponding to active phases. In that case, the `else` branch is wrong, as pu.phase_pos[idx] is not well defined, and in practice we will overwrite the previous active phase there. Perhaps it would be better to not use fixed number of columns for the input data to this function?

(Also minor formatting issue: space around for and else.)